Lib/Src/CUartCom.cpp

```diff
@@ -113,14 +129,38 @@ bool CUartCom::send(etl::string<MAX_STRING_SIZE> msg)
 }
 
 /**
- * @brief Update TX Buffer with next element on queue
+ * @brief Add data to the TX Queue and start transmission if possible
+ * @param Pointer to data buffer
+ * @param Length of the data buffer (in bytes)
+ * @return True if message was added to the queue and transmission started
+ * successfully.
  */
-void CUartCom::updateTxBuffer()
+bool CUartCom::send(uint8_t *p_data_buf, uint32_t len)
 {
-    etl::string<MAX_STRING_SIZE> message = m_tx_queue.get();
-    m_tx_msg_length = message.length();
-    strcpy(m_tx_buffer, message.c_str());
-    m_status = TX;
+    bool b_success = false;
+
+    // Data object to be added to the queue
```

Comment:
superfluous comments. They create an impression that code is well commented, but don't actually contribute to clarity. Try commenting logical segments. 
Example:
```
// Convert data from pointer format to format required by the hardware interface.
...

// Prepare hardware for transmission
...

// Start transmit if not transmitting already

```
This way comments tell the story of what this code does.